src/scheduler/MpiWorld.cpp

```diff
@@ -11,16 +11,12 @@
 #include <faabric/util/macros.h>
 #include <faabric/util/timing.h>
 
-static thread_local std::unordered_map<int, std::thread> asyncThreadMap;
-
 namespace faabric::scheduler {
 MpiWorld::MpiWorld()
   : id(-1)
   , size(-1)
   , thisHost(faabric::util::getSystemConfig().endpointHost)
   , creationTime(faabric::util::startTimer())
-  , threadPool(std::make_shared<faabric::scheduler::MpiAsyncThreadPool>(
```

Comment:
Playing around, I realized that having more or less one worker thread per rank reduces a lot the overhead in managing threads. See the attached flame graphs in the PR description.